src/ixdat/db.py

```diff
@@ -159,36 +161,67 @@ def __init__(self, **self_as_dict):
             setattr(self, attr, value)
         if self_as_dict and not self.column_attrs:
             self.column_attrs = {attr: attr for attr in self_as_dict.keys()}
-        self._backend = MemoryBackend  # SHOULD BE SET AFTER __INIT__ FOR LOADED OBJECT
-        self._id = None  # SHOULD BE SET AFTER THE __INIT__ OF INHERITING CLASSES
+        self._backend = backend or DB.new_object_backend
+        self._id = None  # SHOULD BE SET AFTER THE __INIT__ FOR LOADED OBJECTS
         self.name = None  # MUST BE SET IN THE __INIT__ OF INHERITING CLASSES
 
     def __repr__(self):
-        return f"{self.__class__.__name__}(id={self.id}, name={self.name})"
-
-    @property
-    def backend_name(self):
-        """The name of the backend in which self has been saved to / loaded from"""
-        return self.backend.name
+        return f"{self.__class__.__name__}(id={self.id}, name='{self.name}')"
 
     @property
     def id(self):
         """The principle-key identifier. Set by backend or counted in memory."""
         if not self._id:
-            if self.backend_name == "memory":
-                self._id = memory_backend.get_next_available_id(self.table_name)
+            if self.backend_type in ("none", "memory"):
+                self._id = self.backend.get_next_available_id(self.table_name, obj=self)
+                # TODO: Wouldn't it be better if the backend was always asked for the
+                #   ID by Saveable.__init__ ?
             else:
                 raise DataBaseError(
                     f"{self} comes from {self.backend_name} "
                     "but did not get an id from its backend."
                 )
         return self._id
 
+    @property
+    def identity(self):
+        """identity is the backend if different from the active backend, and the id
```

Comment:
I've refactored it to `short_identity` and changed it to always return the backend and the id.

Reply:
This removes however a way to ensure that all of a measurement's series are saved in the active backend before it gets saved itself (with the different return types that would crash it). So that puts more pressure on not messing up elsewhere...

Reply:
I found a bug resulting from removing the differing return types.
There is no easy way to remove that duality here :( 
The problem is that:
-  `meas.save()` needs to get as the object's serialization a dictionary including `{... "s_ids": [1, 2, 3, ...], ...}`, i.e. `s_ids` etc should be a list of integers representing objects "owned" by `meas`. The location of the "owned" objects is not ambiguous since `meas.save()` will always make sure that they are saved in the active backend before `meas`; BUT
- `Measurement.from_dict(meas.as_dict())`, should make a copy of `meas` which can find the data series even if `meas` has not been saved to the active backend, requiring the backend to be part of the dictionary representation, like `{... "s_ids": [(MemoryBackend, 1), (DirBackend, 3), ...], ...}`, i.e. `s_ids` etc should be tuples of `(backend, id)`. This works via placeholder objects made in `Measurement.__init__` by `fill_object_list`, which checks whether `obj.identity` is a tuple or int. 

I cannot think of a better way to make `as_dict()` give a serialization that satisfies both **lazy copying** (where some "owned" objects are to be found in the MemoryBackend) and **saving** (where "owned" objects all get saved in the same backend and we only want the integer). 

Does this make sense? 
Maybe we need to discuss this?

Reply:
The relevant places in the code:
Where the (backend, id) or just id is made:
https://github.com/ixdat/ixdat/blob/2a2266e35a8c4058d7e821ceef2cd7c16ea2ce31/src/ixdat/db.py#L188
Where the (backend, id) or just id is put into a list
https://github.com/ixdat/ixdat/blob/2a2266e35a8c4058d7e821ceef2cd7c16ea2ce31/src/ixdat/measurements.py#L279
This gets passed to a new object via passed to a new object's __init__ via these methods of the parent class db.Savable:
copying in memory: as_dict() and from_dict(). Owned objects can be from multiple backends, notably including MemoryBackend where from_dict() saves them if they have no other backend.
saving and loading: save() and get(). All owned object are saved in the active backend before the measurement. Only id's since everything is in the same backend.
Where the (backend, id) or just id in that list, after passing to the new object, is put into a place-holder:
https://github.com/ixdat/ixdat/blob/2a2266e35a8c4058d7e821ceef2cd7c16ea2ce31/src/ixdat/measurements.py#L129
And where that place-holder is used to make the new object:
https://github.com/ixdat/ixdat/blob/2a2266e35a8c4058d7e821ceef2cd7c16ea2ce31/src/ixdat/db.py#L379

Reply:
I understand your argument about the country code and the problem with dual purposes. I can't quite wrap my head around the conflict though, so probably, either keep as is, of if it should be changed, it will have to be a discussion at a work shop.

Reply:
Ok to resolve from me.

Reply:
Great. Yes I think to be improved, hopefully clearer after next PR and after the big merge.